z3c/dependencychecker/modules.py

```diff
@@ -0,0 +1,398 @@
+# -*- coding: utf-8 -*-
+from xml.etree import ElementTree
+from z3c.dependencychecker.dotted_name import DottedName
+import ast
+import fnmatch
+import os
+import re
+
+
+TEST_REGEX = r"""
+{0}    # Needs to start with the path separator
+\w*    # There could be some characters (or none)
+test   # has to contain test on it
+s?     # ensure plurals are handled as well
+(      # Needs to end with either a path or a file extension:
+{0}    # path separator
+|      # OR
+\w*    # some possible characters
+\.     # a dot
+\w+    # extension
+)
+""".format(os.sep)
+
+TEST_IN_PATH_RE = re.compile(
+    TEST_REGEX,
+    re.VERBOSE,
+)
+
+
+class BaseModule(object):
+
+    def __init__(self, package_path, full_path):
+        self.path = full_path
+        self._relative_path = self._get_relative_path(package_path, full_path)
+        self.testing = self._is_test_module()
+
+    @staticmethod
+    def _get_relative_path(package_path, full_path):
+        return full_path[len(package_path):]
+
+    def _is_test_module(self):
+        return bool(re.search(TEST_IN_PATH_RE, self._relative_path))
+
+    @classmethod
+    def search_files(cls, top_dir):
+        raise NotImplementedError
+
+    def scan(self):
+        raise NotImplementedError
+
+
+class PythonModule(BaseModule):
+
+    @classmethod
+    def search_files(cls, top_dir):
+        """Finds all python files on the package
+
+        For that it gets the path to where top_level.txt points to,
+        which is not always a folder:
+        - folder example: z3c.dependencychecker itself
+        - file example: flake8-isort or any other single file distribution
+        """
+        if top_dir.endswith('.py'):
+            yield cls(top_dir, top_dir)
+            return
+
+        for path, folders, filenames in os.walk(top_dir):
+            if '__init__.py' not in filenames:
+                # Don't descend further into the tree.
+                # Clear folders variable in-place.
+                folders[:] = []
+                continue
+
+            for filename in filenames:
+                if filename.endswith('.py'):
+                    yield cls(
+                        top_dir,
+                        os.path.join(path, filename),
+                    )
+
+    def scan(self):
```

Comment:
What does this do? Where does it get called?

Sounds like `get_mentioned_dotted_names()` or so?

Reply:
See the large answer some comments above